src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -0,0 +1,308 @@
+    assert(root);
+    assert(strlen(root->name)<strlen(path));
+    assert(strcmp(root->name,path)!=0);
+    assert(strspn(path,root->name) >= strlen(root->name));
+
+    if(!root->child){
+        return NULL;
+    }
+    root=root->child;
+    do{
+        if(!strcmp(root->name,path)) return root;
+        if(strspn(path,root->name) >= strlen(root->name))return find(path,root);
+        if(root->next) root = root -> next;
+    }while(root->next);
+    return NULL;
+}
+
+/******************************************************************************
+** fs Level Methods
+******************************************************************************/
+
+uint32 mount(const char *devname, char *mountpoint){
+    OS_printf("Mount started\n");
+    uint32 i = 0;
+    while(i<MAX_NUM_FS){
+        if(fs[i].devname==NULL);
+        else if(!strcmp(fs[i].devname, devname)){
+            fs[i].mountpoint=mountpoint;
+            return OS_FS_SUCCESS;
+        }
+        i++;
+    }
+    OS_printf("mount failed\n");
+    return OS_FS_ERROR;
+}
+
+uint32 unmount(const char *mountpoint){
+    uint32 i = 0;
+    while(i<MAX_NUM_FS){
+        if(mountpoint==NULL);
+        else if(!strcmp(fs[i].mountpoint, mountpoint)){
+            fs[i].mountpoint=NULL;
+            return OS_FS_SUCCESS;
+        }
+        i++;
+    }
+    return OS_FS_ERROR;
+}
+
+/*
+ * Given a pointer to an FS, inits values and fsobj for root
+ */
+uint32 fs_init(struct FS *filesys, char *devname, char *volname, uint32 blocksize, uint32 numblocks)
+{
+    struct fsobj *o=NULL;
+    if(newfile_get(&o))  return OS_FS_ERROR; //TODO: better error for running out of files
+    if(fsobj_init(o))    return OS_FS_ERROR;
+    filesys->devname        = devname;
+    filesys->mountpoint     = "";
+    filesys->blocksize      = blocksize;
+    filesys->numblocks      = numblocks;
+    filesys->root           = o;
+    filesys->next           = NULL;
+    filesys->prev           = NULL;
+    return OS_FS_SUCCESS;
+}
+
+uint32 newfs_init(char *devname, char *volname, uint32 blocksize, uint32 numblocks)
+{
+    uint32 count = 1, ret = 0;
+    if(!devname)
+        {ret=OS_FS_ERR_INVALID_POINTER;goto err;}
+    if(!blocksize || !numblocks) 
+        {ret=OS_FS_ERROR;goto err;}
+    if(strlen(devname) >= OS_FS_DEV_NAME_LEN || strlen(volname) >= OS_FS_VOL_NAME_LEN)
+        {ret=OS_FS_ERROR;goto err;}
+
+    //the first filesystem is always the tar
+    if(!fs[0].devname){
+        ret = tar_load();
+        if(ret) goto err;
+        ret = fs_init(&fs[0],devname,volname,blocksize,numblocks);
+        if(ret) goto err;
+        return OS_FS_SUCCESS;
+    }
+
+    while(count<MAX_NUM_FS && fs[count].devname){
+        count++;
+    }
+    if(count==MAX_NUM_FS)
+        {ret=OS_FS_ERR_DRIVE_NOT_CREATED;goto err;}
+
+    ret = fs_init(&fs[count],devname,volname,blocksize,numblocks);
+    if(ret) goto err;
+
+    return OS_FS_SUCCESS;
+err:
+    return ret;
+}
```

Comment:
Don't use got here. It makes the whole function confusing... Also fix your brackets!